### PR TITLE
Give Scoped a default equals/hashCode implementation

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

File: main-settings/src/main/scala/sbt/Structure.scala [hunk 1]
```diff
@@ -17,7 +17,18 @@ import sbt.Def.{ Initialize, KeyedInitialize, ScopedKey, Setting, setting }
 import std.TaskExtra.{ task => mktask, _ }
 
 /** An abstraction on top of Settings for build configuration and task definition. */
-sealed trait Scoped { def scope: Scope; val key: AttributeKey[_] }
+sealed trait Scoped {
+  def scope: Scope
+  val key: AttributeKey[_]
+
+  override def equals(that: Any) =
```
Comment (eed3si9n) [inline main-settings/src/main/scala/sbt/Structure.scala]: Wouldn't this create cooperative equality on Scoped subtypes?
If you want equals, I think they should be implemented at the class, not sealed trait.

**Review by eed3si9n — CHANGES_REQUESTED** [review]

File: main-settings/src/main/scala/sbt/Structure.scala (hunk 1 above)
Comment (dwijnand) [reply main-settings/src/main/scala/sbt/Structure.scala]: yes! thank you! I'll change it and add a test that enforces it.

**Comment by dwijnand** [comment]
Travis CI has forgotten about this PR?? closing-and-reopening to wake her up.

**Comment by dwijnand** [comment]
ping?

**Pull request merged**